tsdb/wal.go

```diff
@@ -699,6 +687,12 @@ func (w *SegmentWAL) run(interval time.Duration) {
 			}
 		}
 	}
+	// Drain and process any remaining functions.
+	for f := range w.actorc {
```

Comment:
We can defer this so we can return on `679` line.

Reply:
nice catch, thanks